cmd/ibpacker/main.go

```diff
@@ -0,0 +1,194 @@
+
+		// ami specific
+		awsAccessKeyID     = flag.String("aws-access-key-id", "", "AWS access key ID (required for ami type)")
+		awsSecretAccessKey = flag.String("aws-secret-access-key", "", "AWS secret access key (required for ami type)")
+		awsAmiName         = flag.String("aws-ami-name", "", "destination AMI name (required for ami type)")
+		awsS3Bucket        = flag.String("aws-s3-bucket", "", "S3 bucket (required for ami type)")
+		awsRegion          = flag.String("aws-region", "", "AWS region (required for ami type)")
+	)
+	flag.Parse(args)
+
+	// open SSH connection
+	cfg := ibk.SSHTransportConfig{
+		Host:     *hostname,
+		Username: *username,
+		Timeout:  *connTimeout,
+		Stderr:   os.Stdout,
+	}
+	c, err := ibk.NewSSHTransport(cfg)
+	if err != nil {
+		log.Panic(err)
+	}
+	defer c.Close(ctx)
+
+	// load blueprint into a string
+	blueprint, err := os.ReadFile(*blueprintFile)
+	if err != nil {
+		log.Panic(err)
+	}
+
+	// configure the command
+	cmd := &ibk.ContainerBootCommand{
+		Repository: *repository,
+		Type:       *imageType,
+		Arch:       *arch,
+		Blueprint:  string(blueprint),
+		RootFS:     *rootFS,
+		Common: ibk.CommonArgs{
+			DryRun:      *dryRun,
+			Interactive: *interactive,
+			TTY:         *tty,
+			TeeLog:      *teeLog,
+		},
+	}
+	if *imageType == "ami" {
+		cmd.AWSUploadConfig = &ibk.AWSUploadConfig{
+			AMIName:            *awsAmiName,
+			S3Bucket:           *awsS3Bucket,
+			Region:             *awsRegion,
+			AWSAccessKeyID:     *awsAccessKeyID,
+			AWSSecretAccessKey: *awsSecretAccessKey,
+		}
+	}
+
+	// apply the command
+	err = ibk.ApplyCommand(ctx, cmd, c)
+	if err != nil {
+		log.Panic(err)
+	}
+}
+
+var (
+	hostname    = flag.String("hostname", "", "SSH hostname or IP with optional port (e.g. example.com:22)")
+	username    = flag.String("username", "", "SSH username")
+	dryRun      = flag.Bool("dry-run", false, "dry run")
+	debug       = flag.Bool("debug", false, "debug logging")
+	interactive = flag.Bool("interactive", false, "pass --interactive mode to the container tool")
+	tty         = flag.Bool("tty", false, "pass --tty mode to the container tool")
+	connTimeout = flag.Duration("conn-timeout", 10*time.Second, "SSH connection timeout")
+	timeout     = flag.Duration("timeout", 9999*time.Hour, "transaction timeout (overall build timeout)")
+	teeLog      = flag.Bool("tee-log", true, "tee the output log to a file named build.log")
+)
+
+func main() {
+	ctx, cancel := context.WithTimeout(context.Background(), *timeout)
+	defer cancel()
+
+	flag.Parse()
+	if *interactive || *tty {
+		*teeLog = false
+	}
+
+	level := logutils.LogLevel("WARN")
+	if *debug {
+		level = logutils.LogLevel("DEBUG")
+	}
+	filter := &logutils.LevelFilter{
+		Levels:   []logutils.LogLevel{"DEBUG", "WARN", "ERROR"},
+		MinLevel: level,
+		Writer:   os.Stderr,
+	}
+	log.SetOutput(filter)
+	log.SetFlags(0)
+
+	args := flag.Args()
+	if len(args) == 0 {
+		log.Fatal("Please specify a subcommand.")
```

Comment:
is there an option to show the subcommands on `--help` and maybe list them in this case - or print-help&out :-D

Reply:
Ditto.